app/assets/javascripts/shared/editor_toolbar.js

```diff
@@ -0,0 +1,233 @@
+        $(this).parent().find('[data-btn~=table]').removeClass('disabled');
+      }
+    });
+
+    // when a toolbar button is clicked
+    $('[data-btn]').click(function () {
+      var $element = $(this).parents('[data-behavior~=editor-field]').children('textarea, input[type=text]');
+      var affix = that.affixes[$(this).data('btn')]
+  
+      // inject markdown
+      that.injectMarkdown($element, affix);
+    });
+
+    // keyboard shortcuts
+    $('[data-behavior~=editor-field]').children('textarea, input[type=text]').keydown(function(e) {
+      var key = e.which || e.keyCode; // for cross-browser compatibility
+      if (e.metaKey && key === 66 ) { // 66 = b
+        e.preventDefault();
+        $('[data-btn~=bold]').click();
+      }
+      else if (e.metaKey && key === 73 ) { // 73 = i
+        e.preventDefault();
+        $('[data-btn~=italic]').click();
+      }
+      else if (e.metaKey && key === 75 ) { // 75 = k
+        e.preventDefault();
+        $('[data-btn~=link]').click();
+      }
+    });
+  }
+
+  injectMarkdown($element, affix) {
+    var adjustedPrefixLength = affix.prefix.length,
+        adjustedSuffixLength = affix.suffix.length,
+        startIndex = $element[0].selectionStart,
+        endIndex = $element[0].selectionEnd,
+        elementText = $element.val(),
+        selectedText = $element.val().substring(startIndex, endIndex);
+
+    var markdownText = (startIndex == endIndex) ? affix.asPlaceholder : affix.withSelection(selectedText)
+
+    adjustedPrefixLength *= selectedText.split('\n').length;
+    adjustedSuffixLength *= selectedText.split('\n').length;
+
+    // remove the original selection (if there was one) and add new markdown string in it's place
+    $element.focus(); // bring focus back to $element from the toolbar
+    if (navigator.userAgent.toLowerCase().indexOf('firefox') > -1) { // firefox
+      $element.val(elementText.slice(0, startIndex) + markdownText + elementText.slice(endIndex));
+    }
+    else { // all other browsers
+      document.execCommand('insertText', false, markdownText);
+    }
+    
+    // post-injection cursor location
+    if (startIndex == endIndex) { // no text was selected, select injected placeholder text
+      $element[0].setSelectionRange(startIndex + affix.prefix.length, startIndex + markdownText.length - affix.suffix.length);
+    }
+    else { // text was selected, place cursor after the injected string
+      $element[0].setSelectionRange(adjustedPrefixLength + endIndex + adjustedSuffixLength, adjustedPrefixLength + endIndex + adjustedSuffixLength);
+    }
+
+    // Trigger a change event because javascript manipulation doesn't trigger
+    // them. The change event will reload the preview
+    $element.trigger('textchange');
+  }
+
+  affixesLibrary() {
+    return {
+      'block-code':  new BlockAffix('bc. ', 'Code markup'),
+      'bold':        new Affix('*', 'Bold text', '*'),
+      'header':      new Affix('#[', 'Header text', ']#'),
```

Comment:
```suggestion
      'header':      new Affix('#[', 'Header', ']#'),
```
Suggesting to change this to a single word field name as field names without spaces are a best practice (to allow for filtering if we ever need to)